### ISSUE TITLE
findBy* resolving immediately

### ISSUE THREAD

**Issue by doits** [testing-library/nightwatch-testing-library#80]
I'm trying to get it running with nightwatch and followed the examples, though I cannot get `findByLabelText` to return a promise and wait for the element.

I tried it like this:

```js
const { getQueriesFrom } = require("@testing-library/nightwatch")

describe("test", function() {
  test("other", async function(browser) {
    const { findByLabelText } = getQueriesFrom(browser)

    browser.url("http://localhost:8080")

    const input = await findByLabelText("Not Existent", { exact: false })
  
    console.log(input)
    // =>  
    // {
    //   selector: '.NWTL_findByLabelText-Not-Existent',
    //   nth: [Function: nth],
    //   browser: [Function: browser]
    // }

    browser.setValue(input, "bla@blub.de")
    browser.expect.element(input).value.to.equal("bla@blub.de")
  })
})
```

The problem is that awaiting `findByLabelText` returns immediately and is an object with those keys `selector`, `nth` and `browser`. It does not matter if I try with a not existent label or a label that should exist.

Then the next lines of course do not work and I get this error:

> Expected element <.NWTL_findByLabelText-Not-Existent> to have value equal: "bla@blub.de" - element was not found - expected "equal 'bla@blub.de'" but got: "not present" (5073ms)

and

> NoSuchElementError: An error occurred while running .setValue() command on <.NWTL_findByLabelText-Not-Existent>: undefined; undefined

Shouldn't `findByLabelText` wait until it either finds an input or raise an error if it cannot?

**Comment by doits** [comment]
Alright, I know understand that it should always resolve instantly but return a webdriver selector - so it seems OK that such an object is returned.

So it looks like it is looking to find an element with class `NWTL_findByLabelText-Not-Existent`? Does it mean that? And something should have added the class to the right element in the dom?

**Comment by doits** [comment]
OK, I got it to work:

- looks like `findBy*` is not working, I have to use `getBy*`
- I have to manually pause until the page is fully loaded

```js
const { getQueriesFrom } = require("@testing-library/nightwatch")

module.exports = {
  async other(browser) {
    const { getByLabelText } = getQueriesFrom(browser)

    browser.url("http://localhost:8080")
    browser.pause(5000)

    const input = await getByLabelText("Exists", { exact: false })

    browser.setValue(input, "bla@blub.de")
    browser.expect.element(input).value.to.equal("bla@blub.de")
  }
}
```

This works, though I really don't like the `browser.pause()`. Is there a way to wait until the element becomes available, which I thought `findBy*` would do?

**Comment by benmonro** [comment]
@doits so a thing to keep in mind that is a key difference between NWTL and other testing libraries is that we are translating the queries to CSS selectors (as you saw above).  Have you tried using the queries in conjuection with [waitForElementPresent](https://nightwatchjs.org/api/waitForElementPresent.html)? 


 I also wrote the Testcafe testing library and over there we removed the findBy queries because Testcafe has built in waiting so it doesn't make sense to use findBy in that case.  I'm thinking it might make sense to do the same for NWTL and refer people to waitForElementPresent as there really isn't any way that I'm aware of to have nightwatch wait for an async / promise call inside of a call to `browser.execute` (which is how NWTL injects DTL into the DOM).  I'd love to hear your thoughts if you know of a way to accomplish this.  and of course would welcome a PR! :)
